gemrb/GUIScripts/InventoryCommon.py

```diff
@@ -369,18 +369,21 @@ def OnClose():
 	if GameCheck.IsPST() and slotItem["Flags"] & IE_INV_ITEM_CONVERSABLE:
 
 		drink = True # "Use"
+		
+	def ItemPress(func, *args):
+		return lambda: func(slotItem, *args)
```

Comment:
You could extend OnPress and others in GUIClasses instead, so we can then pass params normally. Almost all our lambda uses are just to work around the fact that we can't pass params.

Reply:
I'm not sure how that would look. how could you know that the function does or doesn't take a control parameter? We could use named parameters I guess, but that is outside the scope of what im doing.

Also, that wouldn't work here since this needs to capture `slotItem`.